src/modules/crypto/mod.rs

```diff
@@ -725,4 +787,52 @@ mod crypto_test {
         // evaluate the result.
         assert_eq!(data2.to_vec(), pt);
     }
+
+    #[test]
+    fn test_rsa_sign_verify() {
+        let mut rsa = RSA::new(Some(2), Some(RSAKeySize::RSA4096)).unwrap();
+        rsa.keygen().unwrap();
+        let data = b"The best way to not feel hopeless is to get up and do something.".to_vec();
+        let sig = rsa.sign(&data).unwrap();
+        let valid = rsa.verify(&data, &sig).unwrap();
+        assert_eq!(valid, true);
+    }
+
+    #[test]
+    fn test_rsa_encrypt_decrypt() {
+        let mut rsa = RSA::new(None, None).unwrap();
+        rsa.keygen().unwrap();
+        let data = b"The best way to not feel hopeless is to get up and do something.".to_vec();
+        let ct = rsa.encrypt(&data).unwrap();
+        let pt = rsa.decrypt(&ct).unwrap();
+        assert_eq!(data, pt);
+    }
+
+    #[test]
+    fn test_ecdsa_sign_verify() {
+        let mut ecdsa = ECDSA::new(None).unwrap();
+        ecdsa.keygen().unwrap();
+        let data = b"The best way to not feel hopeless is to get up and do something.".to_vec();
+        let sig = ecdsa.sign(&data).unwrap();
+        let valid = ecdsa.verify(&data, &sig).unwrap();
+        assert_eq!(valid, true);
+    }
+
+    #[cfg(feature = "crypto_adaptor_tongsuo")]
+    #[test]
+    fn test_sm2_keygen() {
+        assert_eq!(1, 1);
+    }
+
+    #[cfg(feature = "crypto_adaptor_tongsuo")]
+    #[test]
+    fn test_sm2_sign_decrypt() {
+        assert_eq!(1, 1);
+    }
+
+    #[cfg(feature = "crypto_adaptor_tongsuo")]
+    #[test]
+    fn test_sm2_encrypt_decrypt() {
+        assert_eq!(1, 1);
```

Comment:
?

Reply:
This not implemented in the underlying rust-tongsuo crate, so it's not implemented yet.